OptimoveSDK/Sources/Classes/Optimobile/AnalyticsHelper.swift

```diff
@@ -226,17 +232,30 @@ final class AnalyticsHelper {
 
         let path = "/v1/app-installs/\(OptimobileHelper.installId)/events"
 
+        var err : Error? = nil
+        let networkBarrier = DispatchSemaphore(value: 0)
+
         eventsHttpClient.sendRequest(.POST, toPath: path, data: data, onSuccess: { _, _ in
-            if let err = self.pruneEventsBatch(context, eventIds) {
-                print("Failed to prune events batch: " + err.localizedDescription)
-                onSyncComplete?(err)
-                return
-            }
-            self.syncEvents(context: context, onSyncComplete)
+            networkBarrier.signal()
         }) { _, error, _ in
-            print("Failed to send events")
-            onSyncComplete?(error)
+            err = error
+            networkBarrier.signal()
+        }
+
+        networkBarrier.wait()
```

Comment:
Could using `DispatchSemaphore` here might lead to deadlocks? E.g. if the `sendRequest` callbacks are executed on the same `serialQueue` that's being blocked by `networkBarrier.wait()`?

Reply:
It would happen if the network request completion handler would run on the serial queue we are waiting the semaphore on. The completion / error handlers arent run on our private serial queue

Reply:
Got it 👌 thanks for clarifying